module/msg_smt/test/mod_msg_smt_unit_test.c

```diff
@@ -0,0 +1,815 @@
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status =  smt_write_payload(channel_id, fake_offset, NULL, fake_size);
+    TEST_ASSERT_EQUAL(FWK_E_PARAM, status);
+
+    /* Out of range offset: max_payload_size = mailbox_size - header */
+    fake_offset = channel_ctx->max_payload_size;
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_MSG_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = smt_write_payload(channel_id, fake_offset, fake_payload, fake_size);
+    TEST_ASSERT_EQUAL(FWK_E_PARAM, status);
+
+    /* Out of range size: max_payload_size = mailbox_size - header */
+    fake_offset = 0;
+    fake_size = channel_ctx->max_payload_size + 1;
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_MSG_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = smt_write_payload(channel_id, fake_offset, fake_payload, fake_size);
+    TEST_ASSERT_EQUAL(FWK_E_PARAM, status);
+
+    /* Out of range offset + size */
+    fake_offset = 1;
+    fake_size = channel_ctx->max_payload_size;
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_MSG_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = smt_write_payload(channel_id, fake_offset, fake_payload, fake_size);
+    TEST_ASSERT_EQUAL(FWK_E_PARAM, status);
+
+    /* Init payload buffer */
+    for (i = 0; i < FAKE_MAILBOX_SIZE; i++) {
+        fake_payload[i] = i;
+    }
+
+    /* Fill payload */
+    test_reset_fake_msg();
+    fake_offset = 0;
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_MSG_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = smt_write_payload(channel_id, fake_offset, fake_payload, fake_size);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+    TEST_ASSERT_EQUAL(0, fake_msg_out[sizeof(uint32_t) + fake_offset]);
+    TEST_ASSERT_EQUAL(fake_size-1, fake_msg_out[sizeof(uint32_t) + fake_offset + fake_size - 1]);
+
+    /* Fill payload at offset */
+    test_reset_fake_msg();
+    fake_offset = 17;
+    fake_size -= fake_offset;
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_MSG_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = smt_write_payload(channel_id, fake_offset, fake_payload, fake_size);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+    TEST_ASSERT_EQUAL(0, fake_msg_out[sizeof(uint32_t) + fake_offset]);
+    TEST_ASSERT_EQUAL(fake_size-1, fake_msg_out[sizeof(uint32_t) + fake_offset + fake_size - 1]);
+}
+
+static int fake_raise_shm_notification(fwk_id_t channel_id, size_t size)
+{
+    return FWK_SUCCESS;
+}
+
+struct mod_msg_smt_driver_ouput_api fake_mbx_shm_api = {
+    .raise_notification = fake_raise_shm_notification,
+};
```

Comment:
Opening a discussion: I can agree to leave them here as they relate only to the following tests.
but do you think moving them to the top would affect readability? either in +ve/-ve ways.